blocks/footer/footer.js

```diff
@@ -1,487 +1,15 @@
-    const aboutItems = doc.querySelectorAll('div > p:nth-of-type(3) + ul a');
-
-    const about = {
-      title: aboutTitle.textContent,
-      items: toItems(aboutItems),
-    };
-
-    const socialTitle = doc.querySelector('div > p:nth-of-type(4)');
-    const socialItems = doc.querySelectorAll('div > p:nth-of-type(4) + ul a');
-
-    const social = {
-      title: socialTitle.textContent,
-      items: toItems(socialItems),
-    };
-
-    const zeissTitle = doc.querySelector('div > p:nth-of-type(5)');
-    const zeissItems = [
-      doc.querySelectorAll('div > p:nth-of-type(5) + ul a'),
-      doc.querySelectorAll('div > p:nth-of-type(5) + ul + ul a'),
-      doc.querySelectorAll('div > p:nth-of-type(5) + ul + ul + ul a'),
-    ];
-
-    const zeiss = {
-      title: zeissTitle.textContent,
-    };
-    const zeissLocation = path2label(currentPaths[1]);
-    zeissItems.forEach((elements, index) => {
-      zeiss[`items${index}`] = toItems(elements);
-
-      if (zeiss[`items${index}`].some((item) => item.label === zeissLocation)) {
-        zeiss.selected = zeissLocation;
-      }
-    });
-
-    const languageTitle = doc.querySelector('div > p:nth-of-type(6)');
-    const languageGlobalTitle = doc.querySelector('div > p:nth-of-type(7)');
-    const languageEN = doc.querySelector('div > p:nth-of-type(7) + ul a');
-    const languageEuropaTitle = doc.querySelector('div > p:nth-of-type(8)');
-    const languageDE = doc.querySelector('div > p:nth-of-type(8) + ul a');
-
-    const isDE = currentPaths[0] === 'de';
-
-    const language = {
-      title: languageTitle.textContent,
-      selected: isDE ? languageDE.textContent : languageEN.textContent,
-      global: {
-        title: languageGlobalTitle.textContent,
-        // TODO currently hardcoded in the content
-        en: {
-          label: languageEN.textContent,
-          href: languageEN.href,
-        },
-      },
-      europa: {
-        title: languageEuropaTitle.textContent,
-        // TODO currently hardcoded in the content
-        de: {
-          label: languageDE.textContent,
-          href: languageDE.href,
-        },
-      },
-      items: toItems(socialItems),
-    };
-
-    const legalTitle = doc.querySelector('div > p:nth-of-type(9)');
-    const legalItems = doc.querySelectorAll('div > p:nth-of-type(9) + ul a');
-
-    const legal = {
-      title: legalTitle.textContent,
-      items: toItems(legalItems),
-    };
-
-    block.innerHTML = template({
-      nav,
-      relatedSites,
-      popular,
-      about,
-      social,
-      zeiss,
-      language,
-      legal,
-    });
-
-    decorateIcons(block, true);
-
-    addFooterInteractions(block);
-  }
+  const footerUrl = 'https://www.zeiss.com/semiconductor-manufacturing-technology/home.html';
+  const resp = await fetch(footerUrl);
+  const html = await resp.text();
+  const div = document.createElement('div');
+  div.innerHTML = html;
+  const footer = div.querySelector('footer');
+  block.append(footer);
```

Comment:
Since we are just appending the markup, I think the functionality driven by javascript is missing. For e.g. on clicking the dropdown, nothing happens.

Reply:
Good point. IINM the original JS code is contained in a fairly large AEM client library `jupiter-components.all.lc-1.59.0-c51780de-lc.js`.

We could obtain the path to the client by parsing the `footerUrl` response. However it would be wasteful to load the entire 1.4 MB JS library just to enable the footer interraction.

I think we have two options, either (a) have a way to fetch the library specific to the footer, or (b) re-implement the logic.

@sdmcraft wdyt ?